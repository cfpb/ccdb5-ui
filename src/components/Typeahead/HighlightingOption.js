import PropTypes from 'prop-types';
import React from 'react';

export const HighlightingOption = ({ label, position, value }) => {
  if (position < 0) {
    return <span>{label}</span>;
  }

  const start = label.substring(0, position);
  const match = label.substr(position, value.length);
  const end = label.substring(position + value.length);

  return (
    <span>
      {start}
      <b>{match}</b>
      {end}
    </span>
  );
};

HighlightingOption.propTypes = {
  label: PropTypes.string.isRequired,
  position: PropTypes.number.isRequired,
  value: PropTypes.string.isRequired,
};

export default HighlightingOption;
