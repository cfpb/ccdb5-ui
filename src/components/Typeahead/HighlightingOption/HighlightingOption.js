import PropTypes from 'prop-types';

export const HighlightingOption = ({ label, position, value }) => {
  if (position < 0) {
    return <span>{label}</span>;
  }

  const start = label.substring(0, position);
  const match = label.slice(position, position + value.length);
  const end = label.substring(position + value.length);
  return (
    <span>
      {start}
      <b>{match}</b>
      {end}
    </span>
  );
};

HighlightingOption.propTypes = {
  label: PropTypes.string.isRequired,
  position: PropTypes.number.isRequired,
  value: PropTypes.string.isRequired,
};
