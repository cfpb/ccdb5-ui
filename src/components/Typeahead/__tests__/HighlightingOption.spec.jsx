import React from 'react';
import HighlightingOption from '../HighlightingOption';
import renderer from 'react-test-renderer';

describe('component::HighlightingOption', () => {
  describe('initial state', () => {
    it('renders without crashing', () => {
      const target = renderer.create(
        <HighlightingOption label="Foobar" position={0} value="foo" />
      );

      const tree = target.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('handles position == -1', () => {
      const target = renderer.create(
        <HighlightingOption label="Foobar" position={-1} value="foo" />
      );

      const tree = target.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
