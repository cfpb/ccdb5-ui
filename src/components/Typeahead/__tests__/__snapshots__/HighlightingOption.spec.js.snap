// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component::HighlightingOption initial state handles position == -1 1`] = `
<span>
  Foobar
</span>
`;

exports[`component::HighlightingOption initial state renders without crashing 1`] = `
<span>
  
  <b>
    Foo
  </b>
  bar
</span>
`;
